job-runner/chain/traverser.go

```diff
@@ -377,3 +386,36 @@ func (t *traverser) sendJL(pJob proto.Job, err error) {
 		jLogger.Errorf("problem sending job log (%#v) to the RM: %s", jl, err)
 	}
 }
+
+func (t *traverser) retrySequence(j proto.Job) {
+	sequenceStartJob := t.chain.SequenceStartJob(j)
+
+	// Create list of jobs in sequence that have been ran and we need to undo
+	var jobsToUndo []string
+	jobsToUndo = append(jobsToUndo, j.Id)
+	currSeqJobId := sequenceStartJob.Id
+	for currSeqJobId != j.Id {
```

Comment:
Don't we want to include the current job (the one that failed and has prompted the sequence retry) in the jobsToUndo slice?

Reply:
yeah that's done two lines above. it's the first item in the array

Reply:
let me add a code comment to make this more clear